Add assembly.displayName and create AssemblySelector core ui component

cmdcolin opened the pull request:
This proposes creating the assembly.displayName config described in #2183 , and adds an "AssemblySelector" component (something which is duplicated in several places, particularly the import forms)

The assembly selector can then standardize the display of the assemblies' displayName, and code is a little less duplicated

We can consider whether displayName is wanted but I think the standardized assemblyselector is beneficial too

cmdcolin commented:
curious what people think about this one. I think it will be useful because otherwise long assembly names will be repeatedly used throughout the app

if interested, I can add CLI integration with add-assembly, GUI assembly manager, etc. but wanted to get a first pass acceptance of this

garrettjstevens commented:
I think this is a good idea. I'd suggest in the description, maybe instead of "longer" it could be "human readable" or something, since the display name might be shorter.

cmdcolin commented:
Updated the description. Also added to CLI and admin assembly add form

Example screenshot in the admin assembly add form
![Screenshot from 2021-08-20 16-07-45](https://user-images.githubusercontent.com/6511937/130288370-0e88de71-40ff-4a26-8635-b43a1865b194.png)
Example from CLI
![Screenshot from 2021-08-20 16-11-15](https://user-images.githubusercontent.com/6511937/130288418-dca302f6-122c-4cf5-88ba-6cd798d6a0c9.png)
